tests/test_org_sessions_api.py

```diff
@@ -0,0 +1,121 @@
+
+    @pytest.fixture(scope="class")
+    def get_project_id(self, request: object) -> Callable[[], int]:
+        id_iter = itertools.count()
+        next(id_iter)  # skip 0
+        return lambda: next(id_iter)
+
+    def setup_method(self, test_method: Any) -> None:
+        super().setup_method(test_method)
+
+        # values for test data
+        self.minutes = 180
+        self.skew = timedelta(minutes=self.minutes)
+        self.started = datetime.utcnow().replace(
+            minute=0, second=0, microsecond=0, tzinfo=pytz.utc
+        )
+
+        self.storage = get_writable_storage(StorageKey.SESSIONS_RAW)
+
+    def generate_session_events(self, project_id: int) -> None:
+        processor = self.storage.get_table_writer().get_stream_loader().get_processor()
+        meta = KafkaMessageMetadata(
+            offset=1, partition=2, timestamp=datetime(1970, 1, 1)
+        )
+        template = {
+            "session_id": "00000000-0000-0000-0000-000000000001",
+            "distinct_id": "b3ef3211-58a4-4b36-a9a1-5a55df0d9aac",
+            "duration": None,
+            "environment": "production",
+            "org_id": 1,
+            "project_id": project_id,
+            "release": "sentry-test@1.0.1",
+            "retention_days": settings.DEFAULT_RETENTION_DAYS,
+            "seq": 0,
+            "errors": 0,
+            "received": datetime.utcnow().timestamp(),
+            "started": self.started.timestamp(),
+        }
+        events = [
+            processor.process_message(
+                {
+                    **template,
+                    "status": "exited",
+                    "duration": 1947.49,
+                    "session_id": "8333339f-5675-4f89-a9a0-1c935255ab58",
+                    "started": (self.started + timedelta(minutes=13)).timestamp(),
+                },
+                meta,
+            ),
+            processor.process_message(
+                {**template, "status": "exited", "quantity": 5}, meta,
+            ),
+            processor.process_message(
+                {**template, "status": "errored", "errors": 1, "quantity": 2}, meta,
+            ),
+            processor.process_message(
+                {
+                    **template,
+                    "distinct_id": "b3ef3211-58a4-4b36-a9a1-5a55df0d9aac",
+                    "status": "errored",
+                    "errors": 1,
+                    "quantity": 2,
+                    "started": (self.started + timedelta(minutes=24)).timestamp(),
+                },
+                meta,
+            ),
+        ]
+        filtered = [e for e in events if e]
+        write_processed_messages(self.storage, filtered)
+
+    def test_simple(self, get_project_id: Callable[[], int]) -> None:
+        project_id = get_project_id()
+        self.generate_session_events(project_id)
+        project_id2 = get_project_id()
+        self.generate_session_events(project_id2)
+
+        query = Query(
```

Comment:
You're going to need to paginate this query, so I would test it with an orderby.

Reply:
Added an orderby test